File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.java
class Solution {
    public int getLucky(String s, int k) {
        int sum = 0;
        for (int i=0; i<s.length(); i++) {
            int t = (s.charAt(i)-'a'+1);
            if (t > 9) {
                t = getSum(t);
            }
            sum += t;
        }
        k--;
        while (k>0) {
            sum = getSum(sum);
            k--;
        }

        return sum;
    }

    public int getSum(int sum) {
        if (sum == 0) {
            return sum;
        }

        return (sum%10) + getSum(sum/10);
    }
}

File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        sum = 0
        for i in range(len(s)):
            t = ord(s[i])-ord('a')+1
            if (t > 9):
                t = self.getSum(t)
            sum += t
        
        k -= 1
        while (k>0) :
            k-=1
            sum = self.getSum(sum)

        return sum

    def getSum(self, sum):
        if (sum <= 1):
            return sum
        
        return (sum%10) + int(self.getSum(sum/10))

        